standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/CompactionTxnHandler.java

```diff
@@ -1430,6 +1427,44 @@ public void markRefused(CompactionInfo info) throws MetaException {
     updateStatus(info);
   }
 
+
+  @Override
+  public void retryCleanerAttemptWithBackoff(CompactionInfo info, long retentionTime) throws MetaException {
+    try {
+      try (Connection dbConn = getDbConn(Connection.TRANSACTION_READ_COMMITTED)) {
+        try (PreparedStatement stmt = dbConn.prepareStatement("UPDATE \"COMPACTION_QUEUE\" " +
+                "SET \"CQ_TBLPROPERTIES\" = ?, CQ_COMMIT_TIME = ?, CQ_ERROR_MESSAGE= ? "
```

Comment:
CQ_TBLPROPERTIES is for setting TBLPROPERTIES (currently only for MR compaction) like this:

ALTER TABLE table_name [PARTITION (partition_key = 'partition_value' [, ...])]  COMPACT 'compaction_type' WITH OVERWRITE TBLPROPERTIES ("property"="value" [, ...]);

I definitely wouldn't overwrite them for observability reasons. You could add to them, but probably the nicest solution would be to add a new column in the COMPACTION_QUEUE (unfortunately :))

Reply:
Introduced a new dedicated field for this: CQ_RETRY_RETENTION which is non-null and has a default value of 0.